lib/sshutils/sftp/sftp.go

```diff
@@ -275,15 +291,47 @@ func getRemoteHomeDir(sshClient *ssh.Client) (string, error) {
 
 // transfer preforms file transfers
 func (c *Config) transfer(ctx context.Context) error {
+	// get info of source files and ensure appropriate options were passed
+	matchedPaths := make([]string, 0, len(c.srcPaths))
+	fileInfos := make([]os.FileInfo, 0, len(c.srcPaths))
+	for _, srcPath := range c.srcPaths {
+		matches, err := c.srcFS.Glob(ctx, srcPath)
+		if err != nil {
+			return trace.Errorf("error matching glob pattern %q: %w", srcPath, err)
+		}
+		// clean match paths to ensure they are separated by backslashes, as
+		// SFTP requires that
+		for i := range matches {
+			matches[i] = path.Clean(matches[i])
+		}
+		matchedPaths = append(matchedPaths, matches...)
+
+		for _, match := range matches {
+			fi, err := c.srcFS.Stat(ctx, match)
+			if err != nil {
+				return trace.Errorf("could not access %s path %q: %v", c.srcFS.Type(), match, err)
+			}
+			if fi.IsDir() && !c.opts.Recursive {
+				// Note: using any other error constructor (e.g. BadParameter)
```

Comment:
So which one leads to the relogin? BadParameter which is actually returned or anything other than BadParameter (in which case the comment is misleading)?